### PR TITLE
Updated main and MagicWeather README to match purchases-ios

### PR DESCRIPTION
**Pull request by rglanz-rc**
The main README has been updated to match the purchases-ios version. This includes changes to the banner and RevenueCat description at the top and some miscellaneous sections (e.g., SDK Reference, Contributing). The "Installation" section related to React Native has been preserved.

The MagicWeather example app README has been updated to match the purchases-ios version, with instructions for both iOS and Android integration, and screenshots.

Thank you for contributing to react-native-purchases. Before pressing the "Create Pull Request" button, please provide the following:

  - [ x ] A description about what and why you are contributing, even if it's trivial.

  - [ ] The issue number(s) or PR number(s) in the description if you are contributing in response to those.

  - [ ] If applicable, unit tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,29 +34,53 @@ React Native Purchases is a client for the [RevenueCat](https://www.revenuecat.c
...
+
+Or view our React Native sample app:
+- [MagicWeather](examples/MagicWeather)
+
 ## Requirements
 
 The minimum React Native version this SDK requires is `0.64`.
 
+## SDK Reference
+Our full SDK reference [can be found here](https://revenuecat.github.io/purchases-ios-docs).
```
Comment (vegaro) [inline README.md:50]:
```suggestion
Our full SDK reference [can be found here](https://revenuecat.github.io/react-native-purchases-docs/).
```

File: README.md [hunk 2]
```diff
@@ -21,29 +34,53 @@ React Native Purchases is a client for the [RevenueCat](https://www.revenuecat.c
...
+
+## Contributing
+Contributions are always welcome! To learn how you can contribute, please see the [Contributing Guide](./Contributing/CONTRIBUTING.md).
```
Comment (vegaro) [inline README.md:53]: Sadly, we don't have this file in this repo 😢

File: README.md [hunk 3]
```diff
@@ -21,29 +34,53 @@ React Native Purchases is a client for the [RevenueCat](https://www.revenuecat.c
...
-`$ react-native link react-native-purchases`
+### 2. Link library to the project
+
+```
+$ react-native link react-native-purchases
+```
 
 ### Additional iOS Setup
 
-#### If your project uses Cocoapods
```
Comment (vegaro) [inline README.md:46]: This will make the header smaller than "Additional iOS Setup" which is what we want right?

File: README.md [hunk 4]
```diff
@@ -53,35 +90,27 @@ pod 'RNPurchases', :path => '../node_modules/react-native-purchases'
...
-##### Remove Strip Frameworks Phase
+### Remove Strip Frameworks Phase
 During the old manual installation instructions, now deprecated, we indicated to add a build phase to strip fat frameworks. 
 1. In Xcode, in project manager, select your app target.
 2. Open the `Build Phases` tab
 3. Remove the added `Strip Frameworks` phase
 4. Clean `Derived Data` 
 
-##### Link static library
+### Link static library
```
Comment (vegaro) [inline README.md:113]: Same with all these other headers. We want them to be nested, removing these `#` will make them at the same level as  `Additional iOS Setup`

File: examples/MagicWeather/README.md [hunk 5]
```diff
@@ -15,13 +15,84 @@ See minimum react-native version requirements for RevenueCat's *Purchases* SDK [
 
 ## Features
 
-This sample demonstrates:
+| Feature                          | Sample Project Location                   |
+| -------------------------------- | ----------------------------------------- |
+| 🕹 Configuring the *Purchases* SDK  | [App.js](App.js#L17) |
+| 💰 Building a basic paywall         | [src/screens/PaywallScreen/index.js](src/screens/PaywallScreen/index.js) |
+| 🔐 Checking subscription status     | [src/screens/WeatherScreen/index.js](src/screens/WeatherScreen/index.js#L30) |
```
Comment (vegaro) [inline examples/MagicWeather/README.md:22]: Now that I see this, I worry we will forget to update this lines. I think we should probably just point to `WeatherScreen /index.js`?

File: README.md (hunk 3 above)
Comment (rglanz-rc) [reply README.md]: Interesting, the additional #'s were just not nesting any further on my preview, so I removed them.

I have reverted the headers.

File: README.md (hunk 4 above)
Comment (rglanz-rc) [reply README.md]: Reverted.

File: examples/MagicWeather/README.md (hunk 5 above)
Comment (rglanz-rc) [reply examples/MagicWeather/README.md:22]:
```suggestion
| 🔐 Checking subscription status     | [src/screens/WeatherScreen/index.js](src/screens/WeatherScreen/index.js) |
```

**Review by vegaro — APPROVED** [review]

**Pull request merged**